### ISSUE TITLE
seek 65536 not supported with cursor 32768 error while uploading file to server

### ISSUE THREAD

**Issue by sunilthamban** [jyu213/ssh2-sftp-client#59]
Hi,

Versions 
"ssh2-sftp-client": "2.0.1",
"ssh2-utils": "1.0.28"

I am getting a seek error while uploading files to the server resulting in partial uploads.

Here is the code snippet used. I have attached the debug log for your reference.

const Client = require('ssh2-sftp-client')
const fs = require('fs')
const path = require('path')

const co = require('co')

const lSFTPConfig = {
  url: 'xxx.xxx.xxx.xxx',
  port: 4045,
  username: 'username',
  password: 'password',
  // in to the sftp server folder
  to_server_folder: '/inbound',
 
}

function debugLog(debuString)
{
  console.log(debuString)
}

function* getSFTPObject() {
  try {
    const sftp = new Client()
    const connectionObject = {
      host: lSFTPConfig.url,
      port: lSFTPConfig.port,
      username: lSFTPConfig.username,
      password: lSFTPConfig.password,
      debug: debugLog,
      readyTimeout: 100000,
    }
    yield sftp.connect(connectionObject)
    console.log(`SFTP_CONNECTED-${connectionObject.host}:${connectionObject.port}`)
    return sftp
  } catch (e) {
    console.log('SFTP-CONNECT_ERROR', e.message)
    return false
  }
}

function* putPayoutFileToRemoteServer(filename) {
  let sftp = ''
  try {
    console.log(`UPLOADING-${filename}`)
    sftp = yield getSFTPObject()
    yield sftp.put(path.resolve(`${filename}`), `/${lSFTPConfig.to_server_folder}/${filename}`, true)
    console.log(`UPLOADED-REQ-${filename}`)
    return true
  } catch (error) {
    if (sftp) {
      sftp.end()
    }
    console.log('SFTP-PUT-ERROR', error.message)
    return false
  }
}

function* testPut() {
  try {
    yield putPayoutFileToRemoteServer('H2HAPS10388120180327103001.txt')
  } catch (error) {
    console.log(error)
  }
}

co(testPut())

Kindly suggest .

[out.txt](https://github.com/jyu213/ssh2-sftp-client/files/1871477/out.txt)

**Comment by jyu213** [comment]
emm..  the default upload file chunk size is 32768, you can set chunkSize when upload file, just like 'sftp.put(PathString, remotePath, {chunkSize: 65356})'.

**Comment by sunilthamban** [comment]
Hi, 

Setting the chunk size to 65356 did not resolve the problem. it simply gave a seek error at another location. I also checked the issues at ssh2..and got a cue of setting the concurrency to 1 from the default 64 since this issue is with an IBM server. IBM apparently also suggests to disable resume and setting concurrency to 1 apparently does it.

I managed to get it working by using the command in the following manner.

> sftp.get(`${lSFTPConfig.from_server_folder}/${filename}`, true, 'utf-8', { chunkSize: 32768, concurrency: 1 })


Thanks for the suggestion.
